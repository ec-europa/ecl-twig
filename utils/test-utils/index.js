const Twig = require('twig');
const { format } = require('prettier');

const renderTwigFile = (file, options, cb) =>
  Twig.renderFile(file, options, (err, html) =>
    cb(err, format(html, { parser: 'html' }))
  );

const renderTwigFileAsNode = (file, options) =>
  new Promise((resolve, reject) =>
    Twig.renderFile(file, options, (err, html) => {
      if (err) return reject(err);

      const div = document.createElement('div');
      div.innerHTML = html.trim();

      return resolve(div.firstChild);
    })
  );

module.exports = {
  renderTwigFile,
  renderTwigFileAsNode,
};
