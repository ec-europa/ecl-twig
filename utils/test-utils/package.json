{
  "name": "@ecl-twig/test-utils",
  "version": "2.29.0",
  "description": "A local test helper",
  "main": "index.js",
  "private": true,
  "devDependencies": {
    "deepmerge": "4.2.2"
  }
}
