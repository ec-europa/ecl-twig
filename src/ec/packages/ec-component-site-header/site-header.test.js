import path from 'path';
import { renderTwigFileAsNode } from '@ecl-twig/test-utils';

describe('EC - Site Header', () => {
  const template = path.resolve(__dirname, './site-header.html.twig');
  const render = params => renderTwigFileAsNode(template, params);
  const defaultIconPath = 'static/icons.svg';

  describe('Default', () => {
    test('renders correctly', () => {
      expect.assertions(1);

      const data = {
        icon_file_path: defaultIconPath,
        language: {
          url: '/example',
          code: 'en',
          label: 'English',
        },
        header_link: {
          url: '/example-1',
          aria_label: 'European Commission',
        },
        header_image: {
          src: 'static/media/logo--en.30b933cc.svg',
          alt: 'European Commission logo',
          title: 'European Commission',
        },
        search_form: {
          button: {
            label: 'Search',
            icon: {
              type: 'general',
              name: 'search',
              path: defaultIconPath,
              size: 'xs',
            },
          },
        },
      };

      return expect(render(data)).resolves.toMatchSnapshot();
    });
  });

  describe('Translated', () => {
    test('renders correctly', () => {
      expect.assertions(1);

      const data = {
        icon_file_path: defaultIconPath,
        language: {
          url: '/example',
          code: 'fr',
          label: 'Français',
        },
        header_link: {
          url: '/example-1',
          aria_label: 'Commmission Européenne',
        },
        header_image: {
          src: 'static/media/logo--fr.a8aaa7ab.svg',
          alt: 'Commmission Européenne logo',
          title: 'Commmission Européenne',
        },
        search_form: {
          button: {
            label: 'Recherche',
            icon: {
              type: 'general',
              name: 'search',
              path: defaultIconPath,
              size: 'xs',
            },
          },
        },
      };

      return expect(render(data)).resolves.toMatchSnapshot();
    });
  });
});
