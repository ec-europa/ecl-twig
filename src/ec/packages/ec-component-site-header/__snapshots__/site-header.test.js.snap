// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EC - Site Header Default renders correctly 1`] = `
<header
  class="ecl-site-header"
>
  <div
    class="ecl-site-header__container ecl-container"
  >
    <div
      class="ecl-site-header__banner"
    >
      <a
        aria-label="European Commission"
        class="ecl-link ecl-link--standalone"
        href="/example-1"
      >
        <img
          alt="European Commission logo"
          class="ecl-site-header__logo-image"
          src="static/media/logo--en.30b933cc.svg"
          title="European Commission"
        />
      </a>
      <div
        class="ecl-site-header__selector"
      >
        <a
          class="ecl-link ecl-link--standalone"
          data-ecl-language-selector=""
          href="/example"
        >
          English
          <span
            class="ecl-site-header__language-icon"
          >
            <svg
              aria-hidden="true"
              class="ecl-icon ecl-icon--m"
              focusable="false"
            >
              <use
                xlink:href="static/icons.svg#general--language"
              />
            </svg>
            <span
              class="ecl-site-header__language-code"
            >
              en
            </span>
          </span>
        </a>
      </div>
    </div>
    <form
      class="ecl-search-form ecl-site-header__search"
      role="search"
    >
      <div
        class="ecl-form-group ecl-form-group--text-input"
      >
        <input
          class="ecl-text-input ecl-search-form__text-input"
          id="input-search"
          name="search"
          type="text"
        />
      </div>
      <button
        class="ecl-button ecl-button--search ecl-search-form__button"
        type="submit"
      >
        <span
          class="ecl-button__container"
        >
          <span
            class="ecl-button__label"
            data-ecl-label=""
          >
            Search
          </span>
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--xs ecl-button__icon ecl-button__icon--after"
            focusable="false"
          >
            <use
              xlink:href="static/icons.svg#general--search"
            />
          </svg>
        </span>
      </button>
    </form>
  </div>
</header>
`;

exports[`EC - Site Header Default renders correctly with extra attributes 1`] = `
<header
  class="ecl-site-header"
  data-test="data-test-value"
  data-test-1="data-test-value-1"
>
  <div
    class="ecl-site-header__container ecl-container"
  >
    <div
      class="ecl-site-header__banner"
    >
      <a
        aria-label="European Commission"
        class="ecl-link ecl-link--standalone"
        href="/example-1"
      >
        <img
          alt="European Commission logo"
          class="ecl-site-header__logo-image"
          src="static/media/logo--en.30b933cc.svg"
          title="European Commission"
        />
      </a>
      <div
        class="ecl-site-header__selector"
      >
        <a
          class="ecl-link ecl-link--standalone"
          data-ecl-language-selector=""
          href="/example"
        >
          English
          <span
            class="ecl-site-header__language-icon"
          >
            <svg
              aria-hidden="true"
              class="ecl-icon ecl-icon--m"
              focusable="false"
            >
              <use
                xlink:href="static/icons.svg#general--language"
              />
            </svg>
            <span
              class="ecl-site-header__language-code"
            >
              en
            </span>
          </span>
        </a>
      </div>
    </div>
    <form
      class="ecl-search-form ecl-site-header__search"
      role="search"
    >
      <div
        class="ecl-form-group ecl-form-group--text-input"
      >
        <input
          class="ecl-text-input ecl-search-form__text-input"
          id="input-search"
          name="search"
          type="text"
        />
      </div>
      <button
        class="ecl-button ecl-button--search ecl-search-form__button"
        type="submit"
      >
        <span
          class="ecl-button__container"
        >
          <span
            class="ecl-button__label"
            data-ecl-label=""
          >
            Search
          </span>
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--xs ecl-button__icon ecl-button__icon--after"
            focusable="false"
          >
            <use
              xlink:href="static/icons.svg#general--search"
            />
          </svg>
        </span>
      </button>
    </form>
  </div>
</header>
`;

exports[`EC - Site Header Default renders correctly with extra class names 1`] = `
<header
  class="ecl-site-header custom-class custom-class--test"
>
  <div
    class="ecl-site-header__container ecl-container"
  >
    <div
      class="ecl-site-header__banner"
    >
      <a
        aria-label="European Commission"
        class="ecl-link ecl-link--standalone"
        href="/example-1"
      >
        <img
          alt="European Commission logo"
          class="ecl-site-header__logo-image"
          src="static/media/logo--en.30b933cc.svg"
          title="European Commission"
        />
      </a>
      <div
        class="ecl-site-header__selector"
      >
        <a
          class="ecl-link ecl-link--standalone"
          data-ecl-language-selector=""
          href="/example"
        >
          English
          <span
            class="ecl-site-header__language-icon"
          >
            <svg
              aria-hidden="true"
              class="ecl-icon ecl-icon--m"
              focusable="false"
            >
              <use
                xlink:href="static/icons.svg#general--language"
              />
            </svg>
            <span
              class="ecl-site-header__language-code"
            >
              en
            </span>
          </span>
        </a>
      </div>
    </div>
    <form
      class="ecl-search-form ecl-site-header__search"
      role="search"
    >
      <div
        class="ecl-form-group ecl-form-group--text-input"
      >
        <input
          class="ecl-text-input ecl-search-form__text-input"
          id="input-search"
          name="search"
          type="text"
        />
      </div>
      <button
        class="ecl-button ecl-button--search ecl-search-form__button"
        type="submit"
      >
        <span
          class="ecl-button__container"
        >
          <span
            class="ecl-button__label"
            data-ecl-label=""
          >
            Search
          </span>
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--xs ecl-button__icon ecl-button__icon--after"
            focusable="false"
          >
            <use
              xlink:href="static/icons.svg#general--search"
            />
          </svg>
        </span>
      </button>
    </form>
  </div>
</header>
`;

exports[`EC - Site Header Translated renders correctly 1`] = `
<header
  class="ecl-site-header"
>
  <div
    class="ecl-site-header__container ecl-container"
  >
    <div
      class="ecl-site-header__banner"
    >
      <a
        aria-label="Commmission Européenne"
        class="ecl-link ecl-link--standalone"
        href="/example-1"
      >
        <img
          alt="Commmission Européenne logo"
          class="ecl-site-header__logo-image"
          src="static/media/logo--fr.a8aaa7ab.svg"
          title="Commmission Européenne"
        />
      </a>
      <div
        class="ecl-site-header__selector"
      >
        <a
          class="ecl-link ecl-link--standalone"
          data-ecl-language-selector=""
          href="/example"
        >
          Français
          <span
            class="ecl-site-header__language-icon"
          >
            <svg
              aria-hidden="true"
              class="ecl-icon ecl-icon--m"
              focusable="false"
            >
              <use
                xlink:href="static/icons.svg#general--language"
              />
            </svg>
            <span
              class="ecl-site-header__language-code"
            >
              fr
            </span>
          </span>
        </a>
      </div>
    </div>
    <form
      class="ecl-search-form ecl-site-header__search"
      role="search"
    >
      <div
        class="ecl-form-group ecl-form-group--text-input"
      >
        <input
          class="ecl-text-input ecl-search-form__text-input"
          id="input-search"
          name="search"
          type="text"
        />
      </div>
      <button
        class="ecl-button ecl-button--search ecl-search-form__button"
        type="submit"
      >
        <span
          class="ecl-button__container"
        >
          <span
            class="ecl-button__label"
            data-ecl-label=""
          >
            Recherche
          </span>
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--xs ecl-button__icon ecl-button__icon--after"
            focusable="false"
          >
            <use
              xlink:href="static/icons.svg#general--search"
            />
          </svg>
        </span>
      </button>
    </form>
  </div>
</header>
`;

exports[`EC - Site Header Translated renders correctly with extra attributes 1`] = `
<header
  class="ecl-site-header"
  data-test="data-test-value"
  data-test-1="data-test-value-1"
>
  <div
    class="ecl-site-header__container ecl-container"
  >
    <div
      class="ecl-site-header__banner"
    >
      <a
        aria-label="Commmission Européenne"
        class="ecl-link ecl-link--standalone"
        href="/example-1"
      >
        <img
          alt="Commmission Européenne logo"
          class="ecl-site-header__logo-image"
          src="static/media/logo--fr.a8aaa7ab.svg"
          title="Commmission Européenne"
        />
      </a>
      <div
        class="ecl-site-header__selector"
      >
        <a
          class="ecl-link ecl-link--standalone"
          data-ecl-language-selector=""
          href="/example"
        >
          Français
          <span
            class="ecl-site-header__language-icon"
          >
            <svg
              aria-hidden="true"
              class="ecl-icon ecl-icon--m"
              focusable="false"
            >
              <use
                xlink:href="static/icons.svg#general--language"
              />
            </svg>
            <span
              class="ecl-site-header__language-code"
            >
              fr
            </span>
          </span>
        </a>
      </div>
    </div>
    <form
      class="ecl-search-form ecl-site-header__search"
      role="search"
    >
      <div
        class="ecl-form-group ecl-form-group--text-input"
      >
        <input
          class="ecl-text-input ecl-search-form__text-input"
          id="input-search"
          name="search"
          type="text"
        />
      </div>
      <button
        class="ecl-button ecl-button--search ecl-search-form__button"
        type="submit"
      >
        <span
          class="ecl-button__container"
        >
          <span
            class="ecl-button__label"
            data-ecl-label=""
          >
            Recherche
          </span>
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--xs ecl-button__icon ecl-button__icon--after"
            focusable="false"
          >
            <use
              xlink:href="static/icons.svg#general--search"
            />
          </svg>
        </span>
      </button>
    </form>
  </div>
</header>
`;

exports[`EC - Site Header Translated renders correctly with extra class names 1`] = `
<header
  class="ecl-site-header custom-class custom-class--test"
>
  <div
    class="ecl-site-header__container ecl-container"
  >
    <div
      class="ecl-site-header__banner"
    >
      <a
        aria-label="Commmission Européenne"
        class="ecl-link ecl-link--standalone"
        href="/example-1"
      >
        <img
          alt="Commmission Européenne logo"
          class="ecl-site-header__logo-image"
          src="static/media/logo--fr.a8aaa7ab.svg"
          title="Commmission Européenne"
        />
      </a>
      <div
        class="ecl-site-header__selector"
      >
        <a
          class="ecl-link ecl-link--standalone"
          data-ecl-language-selector=""
          href="/example"
        >
          Français
          <span
            class="ecl-site-header__language-icon"
          >
            <svg
              aria-hidden="true"
              class="ecl-icon ecl-icon--m"
              focusable="false"
            >
              <use
                xlink:href="static/icons.svg#general--language"
              />
            </svg>
            <span
              class="ecl-site-header__language-code"
            >
              fr
            </span>
          </span>
        </a>
      </div>
    </div>
    <form
      class="ecl-search-form ecl-site-header__search"
      role="search"
    >
      <div
        class="ecl-form-group ecl-form-group--text-input"
      >
        <input
          class="ecl-text-input ecl-search-form__text-input"
          id="input-search"
          name="search"
          type="text"
        />
      </div>
      <button
        class="ecl-button ecl-button--search ecl-search-form__button"
        type="submit"
      >
        <span
          class="ecl-button__container"
        >
          <span
            class="ecl-button__label"
            data-ecl-label=""
          >
            Recherche
          </span>
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--xs ecl-button__icon ecl-button__icon--after"
            focusable="false"
          >
            <use
              xlink:href="static/icons.svg#general--search"
            />
          </svg>
        </span>
      </button>
    </form>
  </div>
</header>
`;
